docs/blueprints/reputation-store.md

```diff
@@ -0,0 +1,109 @@
+# Audit Cache
+
+## Abstract
+
+Right now, we update the database for every single audit. Whenever we audit a stripe, we get a collection of audit outcomes for nodes holding that stripe (e.g. successes, failures, contained, suspended, etc...). For each outcome, we make a database transaction and update all the nodes for that audit outcome at once. 
+
+This is not very performant. The `nodes` table is also heavily contested, and is one of the most crucial pieces for performant network operations - it is fundamental for uploads and downloads, and any delay will be directly seen by customers.
+
+This document describes a solution for an interface to a central reputation store, which occasionally syncs with the `nodes` table. This allows us to transfer the workload of audits off of the `nodes` table to another store. 
+
+## Background
+
+The reason the `nodes` table has such poor performance is because we are doing a lot of row updates (e.g. audits, checkins) at the same time as we are trying to scan over the entire table (e.g. node selection, redash). Independently, row updates or scans work very well on Cockroach, but when combined, they can result in very undesirable performance.
+
+Having a separate store for reputation allows the audit service to make lots of updates for individual nodes, without needing to worry about bottlenecks caused by queries which need to do scans of the `nodes` table.
+
+Some fields are necessary in both the `nodes` table and the new reputation store. We will call these "contested" columns. Some examples are `disqualified` and `suspended`. Node selection needs to read them frequently, and audit needs to update them depending on audit outcomes.
```

Comment:
Note: if we end up collapsing the contested columns in the nodes table to 'healthy', this is unnecessary

Reply:
added to the end of the section: "Note that we may replace the contested values in the overlay with a single `healthy` column (see "Other Details" below). "